allennlp/modules/seq2seq_encoders/pytorch_seq2seq_wrapper.py

```diff
@@ -14,33 +14,50 @@ class PytorchSeq2SeqWrapper(Seq2SeqEncoder):
     want to, e.g., define a linear + softmax layer on top of this to get some distribution over a
     set of labels.  The linear layer needs to know its input dimension before it is called, and you
     can get that from ``get_output_dim``.
+
+    In order to use this API wrapper, you must implement a class with the following interface:
+    Class:
+        self.input_size
+        self.hidden_size
+        self.bidirectional (optional)
+
+    def forward(inputs: PackedSequence, hidden_state: torch.Tensor)
+        -> outputs: PackedSequence, final_state: torch.Tensor.
     """
     def __init__(self, module: torch.nn.modules.RNNBase) -> None:
         super(PytorchSeq2SeqWrapper, self).__init__()
         self._module = module
-        if not self._module.batch_first:
-            raise ConfigurationError("Our encoder semantics assumes batch is always first!")
+        try:
+            if not self._module.batch_first:
+                raise ConfigurationError("Our encoder semantics assumes batch is always first!")
+        except AttributeError:
+            pass
 
     def get_input_dim(self) -> int:
         return self._module.input_size
 
     def get_output_dim(self) -> int:
-        return self._module.hidden_size * (2 if self._module.bidirectional else 1)
+        try:
+            is_bidirectional = self._module.bidirectional
+        except AttributeError:
+            is_bidirectional = False
+        return self._module.hidden_size * (2 if is_bidirectional else 1)
 
     def forward(self,  # pylint: disable=arguments-differ
                 inputs: torch.Tensor,
-                sequence_lengths: torch.LongTensor = None) -> torch.Tensor:
+                sequence_lengths: torch.LongTensor = None,
+                hidden_state: torch.Tensor = None) -> torch.Tensor:
 
         if sequence_lengths is None:
-            return self._module(inputs)[0]
+            return self._module.forward(inputs, hidden_state)[0]
```

Comment:
yes, it will. I just pushed another commit to that fix addressing their concerns, but we'll still need to use a particular commit until the next release. Or just call `forward` and don't use any hooks.